src/lib/detection_algorithms/cntk/exporter.js

```diff
@@ -0,0 +1,158 @@
+    //  tags - a list of objects containing the tagging data. Each object is in the format of:
+    //         {'x1' : int, 'y1' : int, 'x2' : int, 'y2' : int, 'class' : string}
+    //         Where (x1,y1) and (x2,y2) are the coordinates of the top left and bottom right corner
+    //         of the bounding boxes (respectively), and 'class' is the name of the class.
+    // Returns: A Promise object that resolves when the operation completes
+    this.exportFrame = function exportFrame(frameFileName, frameBuffer, tags) {
+        return new Promise(function(resolve, reject) {
+            async.waterfall([
+                deleteFileIfExists.bind(null, path.join(self.negDirPath,  frameFileName)),
+                deleteFileIfExists.bind(null, path.join(self.posDirPath,  frameFileName)),
+                deleteFileIfExists.bind(null, path.join(self.testDirPath, frameFileName)),
+                deleteFileIfExists.bind(null, path.join(self.posDirPath,  `${frameFileName}.bboxes.labels.tsv`)),
+                deleteFileIfExists.bind(null, path.join(self.testDirPath, `${frameFileName}.bboxes.labels.tsv`)),
+                deleteFileIfExists.bind(null, path.join(self.posDirPath,  `${frameFileName}.bboxes.tsv`)),
+                deleteFileIfExists.bind(null, path.join(self.testDirPath, `${frameFileName}.bboxes.tsv`)),
+                function determineWritePath(cb){
+                    if( !tags.length){ 
+                        cb(null, self.negDirPath);
+                    } else {
+                         if (self.testFrameIndecies.includes(self.posFrameIndex)){
+                             self.posFrameIndex++;
+                             cb(null, self.testDirPath);
+                         } else {
+                             self.posFrameIndex++;
+                             cb(null, self.posDirPath);
+                         }
+                    }
+                },
+                function saveImage(exportPath, cb) {
+                    var imageFilePath  = path.join(exportPath, frameFileName);
+                    fs.writeFileSync(imageFilePath, frameBuffer);
+                    cb(null, exportPath); 
+                },
+                function saveBboxes(exportPath, cb) {
+                    if (exportPath !== self.negDirPath){
+                        var labelData = '';
+                        var bboxesData = '';
+                        for (var i in tags) {
+                            var tag = tags[i];
+                            labelData  += `${tag.class}\n`;
+                            bboxesData += `${tag.x1}\t${tag.y1}\t${tag.x2}\t${tag.y2}\n`;
+                        }
+                        var labelFilePath  = path.join(exportPath, `${frameFileName}.bboxes.labels.tsv`);
+                        var bboxesFilePath  = path.join(exportPath, `${frameFileName}.bboxes.tsv`);
+                        fs.writeFileSync(labelFilePath, labelData);
```

Comment:
You can just have the callback of the first trigger the second write. If you want to get fancier you can use the async module (you can use the parallel method, or forEach, or anything that work)